Library/Homebrew/linkage_checker.rb

```diff
@@ -152,9 +155,18 @@ def check_dylibs(rebuild_cache:)
     checked_dylibs = Set.new
 
     keg_files_dylibs.each do |file, dylibs|
+      file_has_any_rpath_dylibs = T.let(false, T::Boolean)
       dylibs.each do |dylib|
         @reverse_links[dylib] << file
 
+        # Binary executables that link @rpath-prefixed dylibs must include at
+        # least one rpath in order to resolve it.
+        if !file_has_any_rpath_dylibs && (dylib.start_with? "@rpath/")
+          file_has_any_rpath_dylibs = true
+          pathname = Pathname(file)
+          @executables_missing_rpaths << file if pathname.binary_executable? && pathname.rpaths.empty?
+        end
```

Comment:
Why not include this in the `if dylib.start_with? "@"` block?

Reply:
We want to find the full set of binaries that are missing rpaths. Moving this to that block would only examine the first binary to reference a given dylib.